shared/profile/search-container.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import {onUserClick} from '../actions/profile'
+import * as Creators from '../actions/chat/creators'
+import * as SearchCreators from '../actions/searchv3/creators'
+import * as SearchConstants from '../constants/searchv3'
+import {debounce} from 'lodash'
+import {compose, withState, withHandlers, defaultProps} from 'recompose'
+import {connect} from 'react-redux'
+import {profileSearchResultArray} from '../constants/selectors'
+import Search from './search'
+
+import type {Props} from './search'
+import type {TypedState} from '../constants/reducer'
+
+type HocIntermediateProps = {
+  _clearSearchResults: () => void,
+  _onClick: (id: string) => void,
+  _onChangeText: (nextText: string) => void,
+  _onSelectService: () => void,
+  _search: (term: string, service: SearchConstants.Service) => void,
+}
+
+const mapStateToProps = (state: TypedState) => ({
+  ids: profileSearchResultArray(state),
+})
+const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, onBack, onToggleSidePanel}: Props) => ({
+  _clearSearchResults: id => dispatch(Creators.clearSearchResults()),
```

Comment:
:sheep: Is `id` here really a parameter? I expected this to have no arguments according to the type.